RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -335,6 +335,12 @@ class PumpOpsSynchronous {
 
         var events = [TimestampedHistoryEvent]()
         var timeAdjustmentInterval: NSTimeInterval = 0
+        
+        // Going to scan backwards in time through events, so event time should be monotonically decreasing
+        // Start with some time in the future, to account for the condition when the pump's clock is ahead
+        // of ours by a small amount.
+        var timeCursor = NSDate(timeIntervalSinceNow: NSTimeInterval(minutes: 60))
+        let eventTimestampDeltaAllowance = NSTimeInterval(minutes: 60)
```

Comment:
It should be totally fine to say event start dates go backward within 1 hour if you test the scheduled event entry cases: BasalProfileStart/Sarah6e while a square is running. As long as the square isn't re-written as its delivered, after those events, then you're OK.